File: lib/api/members.js
var _ = require('lodash'),
  Promise = require('bluebird'),
  util = require('../util'),
  log = util.logger();

function Members(hootsuite, connection) {
  this._hootsuite = hootsuite;
  this._connection = connection;
}

Members.prototype = {
  findById: function (memberId) {
    var path = util.createPath('members', memberId);
    return this._connection.get(path, { _method: 'GET' });
  },
  create: function (msg) {
    var path = util.createPath('members');
    var data = {
      organizationIds: msg.organizationIds || [],
      email: msg.email || null,
      fullName: msg.fullName || undefined,
      companyName: msg.companyName || undefined,
      bio: msg.bio || undefined,
      timezone: msg.timezone || undefined,
      language: msg.language || undefined,
    };
    return this._connection.postJson(path, data);
  },
  findByIdOrgs: function (memberId) {
    var path = util.createPath('members', memberId, 'organizations');
    return this._connection.get(path, { _method: 'GET' });
  },
}

module.exports = Members;

File: src.net/Hootsuite.Rest/Api/Members.cs
using Newtonsoft.Json.Linq;
using System.Threading.Tasks;

namespace Hootsuite.Rest.Api
{
    public class Members
    {
        Hootsuite _hootsuite;
        Connection _connection;

        public Members(Hootsuite hootsuite, Connection connection)
        {
            _hootsuite = hootsuite;
            _connection = connection;
        }

        public Task<JObject> findById(string memberId)
        {
            var path = util.createPath("members", memberId);
            return _connection.get(path);
        }

        public Task<JObject> create(dynamic msg)
        {
            var path = util.createPath("members");
            var data = new
            {
                organizationIds = dyn.getProp(msg, "organizationIds", new string[0]),
                email = dyn.getProp(msg, "email", (string)null),
                fullName = dyn.getProp(msg,"fullName", (string)null),
                companyName = dyn.getProp(msg, "companyName", (string)null),
                bio = dyn.getProp(msg, "bio", (string)null),
                timezone = dyn.getProp(msg, "timezone", (string)null),
                language = dyn.getProp(msg, "language", (string)null),
            };
            return _connection.postJson(path, data);
        }

        public Task<JObject> findByIdOrgs(string memberId)
        {
            var path = util.createPath("members", memberId, "organizations");
            return _connection.get(path);
        }
    }
}
